packages/truffle-core/lib/command.js

```diff
@@ -100,6 +100,35 @@ class Command {
       }
     });
 
+    //Check unsupported command line flag according to the option list in help
+    try {
+      const inputOptions = [];
+      inputStrings.map(i => {
+        if (i.startsWith("--")) {
+          inputOptions.push(i);
+        }
+      });
+
+      const validOptions = [];
+      result.command.help.options.map(item => {
+        let opt = item.option.split(" ")[0];
+        if (opt.startsWith("--")) validOptions.push(opt);
+      });
+
+      let notValidOptions = inputOptions.filter(
+        opt => !validOptions.includes(opt)
+      );
+
+      if (notValidOptions.length > 0)
+        return callback(
+          new TaskError(
+            "Unsupported (Undocumented) command line option: " + notValidOptions
+          )
+        );
```

Comment:
Hmm, so if you are up for it I think we can extend this `try/catch` and encapsulate all the code from here through the next `try/catch` since the catch's do the same thing. In other words, something like
```
      if (notValidOptions.length > 0)
        return callback(
          new TaskError(
            "Unsupported (Undocumented) command line option: " + notValidOptions
          )
        );

      const newOptions = Object.assign({}, clone, argv);

      result.command.run(newOptions, callback);
      analytics.send({
        command: result.name ? result.name : "other",
        args: result.argv._,
        version: bundled || "(unbundled) " + core
      });
    } catch (err) {
      callback(err);
    }
```
I don't think there is any reason we want to strictly not include line 132 from it.